File: backend/internal/models/review.go
package models

import (
	"time"
)

type Review struct {
    ID          string    `json:"id"`           // Unique identifier
    AppID       string    `json:"app_id"`       // iTunes app ID
    Author      string    `json:"author"`
    Content     string    `json:"content"`
    Rating      int       `json:"rating"`       // Score (1-5)
    SubmittedAt time.Time `json:"submitted_at"`
    FetchedAt   time.Time `json:"fetched_at"`   // When we fetched it
}

File: frontend/src/types/review.ts
export interface Review {
  id: string;
  app_id: string;
  author: string;
  content: string;
  rating: number;
  submitted_at: string;
  fetched_at: string;
}

export interface AverageRating {
  app_id: string;
  average_rating: number;
  review_count: number;
  hours: number;
}